roles/nvidia-docker/tasks/main.yml

```diff
@@ -0,0 +1,73 @@
+- name: "Install GCCv10"
+  package:
+    name: gcc10.x86_64
+    state: present
+
+- name: "Remove nouveau kernel module"
+  modprobe:
+    name: nouveau
+    state: absent
+
+- name: "Disable nouveau kernel module"
+  copy:
+    dest: /usr/lib/modprobe.d/nvidia-installer-disable-nouveau.conf
+    content: |
+      blacklist nouveau
+      options nouveau modeset=0
+
+- name: "Download nvidia-{{ nvidia_driver_version }} driver"
+  get_url:
+    url: "https://us.download.nvidia.com/XFree86/Linux-x86_64/{{ nvidia_driver_version }}/NVIDIA-Linux-x86_64-{{ nvidia_driver_version }}.run"
+    dest: "/root/NVIDIA-{{ nvidia_driver_version }}.run"
+
+- name: "Install nvidia-{{ nvidia_driver_version }} driver"
+  shell: "sh /root/NVIDIA-{{ nvidia_driver_version }}.run --silent"
+  register: nvidia_install_log
+  environment:
+    CC: "/bin/gcc10-gcc"
+    CXX: "/bin/gcc10-gcc"
+
+- name: "Install nvidia-{{ nvidia_driver_version }} log"
+  debug: 
+    msg: "{{ nvidia_install_log.stdout_lines }}"
+
+- name: "nvidia-docker repo"
+  yum_repository:
+    name: "nvidia-docker-{{ item.split('/')[3] }}-{{ item.split('/')[4] }}"
+    description: "nvidia-docker-{{ item.split('/')[3] }}-{{ item.split('/')[4] }}"
+    baseurl: "{{ item }}"
+  loop:
+    - "https://nvidia.github.io/libnvidia-container/experimental/amzn2/$basearch"
+    - "https://nvidia.github.io/libnvidia-container/stable/amzn2/$basearch"
+    - "https://nvidia.github.io/nvidia-container-runtime/experimental/amzn2/$basearch"
+    - "https://nvidia.github.io/nvidia-container-runtime/stable/amzn2/$basearch"
+    - "https://nvidia.github.io/nvidia-docker/amzn2/$basearch"
+
+- name: "Install nvidia-docker2"
+  yum:
+    update_cache: yes
+    name: "{{ item }}"
+    state: present
+    disable_gpg_check: yes # TODO: import keys
+  loop:
+    - nvidia-container-runtime
+    - nvidia-docker2
+
+- name: "Restart docker"
+  systemd:
+    name: docker
+    state: restarted
+
```

Comment:
At the end of this, can we delete the nvidia driver install file since it's rather large?